src/DRICH_geo.cpp

```diff
@@ -546,29 +574,94 @@ static Ref_t createDetector(Detector& desc, xml::Handle_t handle, SensitiveDetec
+	    sensorDE.setPlacement(sensorPV);
+	    if (!debugOptics || debugOpticsMode == 3 || debugOpticsMode == 5) {
+	      SkinSurface sensorSkin(desc, sensorDE, "sensor_optical_surface_" + modsecName, sensorSurf, sensorVol);
+	      sensorSkin.isValid();
+	    };
+
+	    // increment sensor module number
+	    imod++;
+
+	  }; // end patch cuts
+	};   // end phiGen loop
+      };     // end thetaGen loop
+    }; // end sensor loop
+
+    // large sensor placement (for focal point testing):
+    if (debugOpticsMode == 4){
+      Box    sensorSolidScaled(sensorRescale * sensorSide / 2., sensorRescale * sensorSide / 2., sensorThickness / 2.);
+      Volume sensorVolScaled(detName + "_sensor_" + secName, sensorSolidScaled, sensorMat);
+      sensorVolScaled.setVisAttributes(sensorVis);
+      sensorVolScaled.setSensitiveDetector(sens);
+
+      double phiGen = 0;
+      auto sensorPlacement =
+        RotationZ(sectorRotation) *
+        Translation3D(sensorSphPos.x()+100, sensorSphPos.y(), sensorSphPos.z()-125) *
+        RotationX(phiGen) *
+        Translation3D(sensorSphRadius, 0., 0.)*
+        RotationZ(0);
+      auto sensorPV = gasvolVol.placeVolume(sensorVolScaled, sensorPlacement);
+      sensorPV.addPhysVolID("sector", isec).addPhysVolID("module", imod);
+
+      DetElement sensorDE(det, Form("sensor_de%d_%d", isec, imod), 0);
+      sensorDE.setPlacement(sensorPV);
+
+      SkinSurface sensorSkin(desc, sensorDE, Form("sensor_optical_surface%d", isec), sensorSurf, sensorVolScaled);
+      sensorSkin.isValid();
+    }; // end large sensor placement
+
+    // for debugOpticsMode 5: drawing calculated focal points
+    // of thrown beams of parallel photons
+    if( std::getenv("FPPLOT_FILE")!=NULL && debugOpticsMode == 5){
+      //std::cout << "drawing focal points " << std::endl;
+      Cone FPsolid(1.,0.25,0.5,1,1);
+      Volume FPvol(detName + "_FPpos_" + secName, FPsolid, aerogelMat);
+      FILE * fptxt = fopen(std::getenv("FPPLOT_FILE"),"r");
+      double fpx, fpy, fpz, dirx, diry, dirz;
+      int fpnum=0;
+
+      while(fscanf(fptxt,"%lf %lf %lf %lf %lf %lf", &fpx, &fpy, &fpz, &dirx, &diry, &dirz)!=EOF){
+        if( std::abs(fpx) < 1000  && std::abs(fpy) < 1000 && std::abs(fpz) < 1000){
+          double zrot = 0;
+          if(dirx < 0){
+            if(diry < 0) zrot = -1*std::atan2(dirx,diry) + M_PI;
+            else zrot = -1*std::atan2(dirx,diry);
+          }
+          else{
+            if(diry < 0) zrot = -1*std::atan2(dirx,diry) - M_PI;
+            else zrot = std::atan2(dirx,diry) + M_PI;
+          }
+          double xrot = std::atan2(sqrt(diry*diry+dirx*dirx),abs(dirz));
+	  xrot = 0;
+	  zrot = 0;
+          auto FPPlacement =
+            RotationZ(0) *
+            Translation3D(fpx, fpy, fpz-vesselPos.z())*
+            RotationX(xrot)*
+            RotationZ(zrot)
+            ;
+          auto FPPV = gasvolVol.placeVolume(FPvol, FPPlacement);
```

Comment:
Variable names which are all-caps are considered reserved for preprocessor definitions, absolute constants, and environment variable names.